app/models/manageiq/providers/vmware/infra_manager/inventory/collector/property_collector.rb

```diff
@@ -171,5 +171,7 @@ def get_array_entry(array, key)
         return array, n if entry_key.to_s == key
       end
     end
+
+    return nil, nil
```

Comment:
So turned out this wasn't finding anything but didn't return `nil` so was bypassing the existing check [[here]](https://github.com/ManageIQ/manageiq-providers-vmware/pull/739/files#diff-92f8567cb0213c47e120eb29c584720ecf8c8867efdb142ad6718c6f26c30ae0R111)